postgres/datadog_checks/postgres/config.py

```diff
@@ -2,322 +2,467 @@
+
+    def add_error(self, error: str | ConfigurationError):
+        """
+        Add an error to the validation result.
+        :param error: The error message to add.
+        """
+        self.errors.append(ConfigurationError(error) if isinstance(error, str) else error)
+        self.valid = False
+
+    def add_warning(self, warning: str):
+        """
+        Add a warning to the validation result.
+        :param warning: The warning message to add.
+        """
+        self.warnings.append(warning)
+
+
+def build_config(check: PostgreSql) -> Tuple[InstanceConfig, ValidationResult]:
+    """
+    Build the Postgres configuration.
+    :param check: The check instance.
+    :param init_config: The init_config for the Postgres check.
+    :param instance: The instance configuration for the Postgres check.
+    :return: InstanceConfig
+        The instance configuration object.
+    """
+
+    instance = check.instance
+    init_config = check.init_config
+
+    args = {}
+
+    # Automatically set values that support defaults or that have simple values in the instance
+    instance_config_fields = set(InstanceConfig.__annotations__.keys())
+    from datadog_checks.postgres.config_models import defaults
+
+    for f in instance_config_fields:
+        try:
+            args[f] = getattr(defaults, f"instance_{f}")()
+        except AttributeError:
+            args[f] = None
+        if f in instance:
+            args[f] = instance[f]
+
+    from datadog_checks.postgres.config_models import dict_defaults
+
+    # Set values for args that have deprecated fallbacks, are not supported by the spec model
+    # or have other complexities
+    # If you change a literal value here, make sure to update spec.yaml
+    args.update(
+        {
+            # Set the default port to None if the host is a socket path
+            "port": instance.get('port', instance_port() if not instance.get('host', '').startswith('/') else None),
+            # Set None values for ssl
+            # Database configuration
+            "dbm": instance.get(
+                'dbm', instance.get('deep_database_monitoring', instance_dbm())
+            ),  # Deprecated, use `dbm` instead
+            "custom_metrics": map_custom_metrics(
+                instance.get('custom_metrics', [])
+            ),  # Deprecated, use `custom_queries` instead
+            "custom_queries": instance.get('custom_queries', []),
+            "database_identifier": instance.get('database_identifier', {"template": "$resolved_hostname"}),
+            "database_autodiscovery": {
+                **dict_defaults.instance_database_autodiscovery(),
+                **(instance.get('database_autodiscovery', {})),
+            },
+            "query_metrics": {
+                **dict_defaults.instance_query_metrics(),
+                **{
+                    "batch_max_content_size": init_config.get('metrics', {}).get('batch_max_content_size', 20_000_000),
```

Comment:
Can `batch_max_content_size` be added into `dict_defaults.instance_query_metrics`. It would be good if we can avoid having config constants scattered around